controllers/routemap.go

```diff
@@ -216,6 +218,36 @@ versionLoop:
 	return versionsAvailable
 }
 
+// computeSignature computes and sets the signature for a particular version
+// signature is based on the spec section
+func computeSignature(v version) (uint64, error) {
+	// get all resources of a version
+	resources := []interface{}{}
+	for _, resource := range v.Resources {
+		obj, err := appInformers[resource.GVRShort].Lister().ByNamespace(*resource.Namespace).Get(resource.Name)
+		if err != nil {
+			return 0, fmt.Errorf("cannot get resource: %e", err)
+		}
+
+		// extract spec section from resource, if applicable
+		specSection, _, err := unstructured.NestedFieldNoCopy(obj.(*unstructured.Unstructured).Object, spec)
+		if err != nil {
+			// error here does not mean no spec, it means cannot traverse the object (to find if there is a spec)
+			return 0, fmt.Errorf("cannot traverse resource: %e", err)
+		}
+
+		resources = append(resources, specSection)
```

Comment:
Is the length of `resources` always guaranteed to be equal to length of `v.Resources`? This is desirable.

In particular, if some objects don't have spec, it looks like some nil values would be added to `resources` ... correct?

Reply:
Yes, `nil` should be added.